### ISSUE TITLE
LR7 - new Yo/Gulp theme deployment toolchain breaks at server hot deploy

### ISSUE THREAD

**Issue by Jwittkopp** [liferay/liferay-theme-tasks#26]
Hi there, 

tried your new LR7 toolchain for theme development based on Yo / Gulp.

Somehow it seems to break at the point of hot deploy on my target server (regardless of whether the server is running whilst deploying the theme or whether I fire up the server after Gulp has copied the files to the /deploy directory). I did read the health warning on NPM that server-manager-web plugin has not yet been migrated to 7.0.

Logs and settings attached.

[Install trace 07-06-2016.pdf](https://github.com/liferay/liferay-theme-tasks/files/306237/Install.trace.07-06-2016.pdf)

**Comment by Robert-Frampton** [comment]
Hi @Jwittkopp 

Thanks for the detailed description. By the looks of the logs you provided this took place after the watch task was used correct? If the server was shutdown during a watch task and then restarted, you'll have to deploy your theme with the server running to properly teardown the watch task.

Can you confirm that a `gulp deploy` fails when you have your server running?

**Comment by Jwittkopp** [comment]
Hi Robert,


firstly, thanks for your prompt reply.


I originally used gulp watch following Marcos' and Nate's explanations in their DevCon videos, but then realised that I had to install and get a gogo shell up and running.


I have since reinstalled yo and gulp, as well as the generator-liferay-theme, and only really run gulp build and gulp deploy commands at the moment. I have gone through the logs step-by-step again now and think I have fixed the problem now.


It's down to issues of access permissions, where specifically npm and yo appear to be very sensitive against. For example, some directories that are only being used temporarily are not assigned proper ownership, which somewhat throws the following install process without properly alerting the user i.e. your prefix directive in the .npmrc documentation may not be sufficient.


I understand that these are not Liferay SW packages. Maybe some health warning in the "Themes and Templates" section of your documentation could be useful for others to not experience the same hiccups.


Other than that, it really works nicely. I am using this in combination with Expresso for theme development.


All the best, Jurgen


________________________________
